frontend/src/pages/CategoryPage.tsx

```diff
@@ -1,8 +1,49 @@
+import Categories from "../components/Categories";
+import {FormEventHandler, useContext, useEffect, useState} from "react";
+import {ICategoryController} from "../models/ControllerTypes";
+import CategoryController from "../controllers/CategoryController";
+import {Category} from "../models/Category";
+import {AuthContext} from "../context/AuthProvider";
+
+interface ITextInput {
+    categoryInput: { value: string }
+}
 
 export default function CategoryPage() {
+    const config = useContext(AuthContext).config!
+    console.log(config)
+    const controller: ICategoryController = CategoryController(config);
+    const [categories, setCategories] = useState<Category[]>([])
+
+    useEffect(() => {
+        if (!config) return
+        controller.getCategories().then(setCategories)
+        //eslint-disable-next-line
+    }, [config])
+
+    const addCategory: FormEventHandler<HTMLFormElement> = (event) => {
+        event.preventDefault()
+        const form = event.currentTarget
+        const formElements = form.elements as typeof form.elements & ITextInput
+        const category = formElements.categoryInput.value.trim()
+
+        if (category && category.length > 0) {
+            controller.getCategories().then(setCategories)
+        }
```

Comment:
I would create a state for the input field. Then you have full controll over the input and getting the value is a bit easier